### PR TITLE
tests: allow bypass keyword for firewall rules - v3

### PR DESCRIPTION
**Pull request by jufajardini**
Previous: #3257 
- add tests for corner case scenarios with td rules with packet drops
- add tests for `pre_flow` and `pre_stream` special hooks -- these represent current behavior, not sure if we'll want to merge them or not

## Ticket

Redmine ticket: https://redmine.openinfosecfoundation.org/issues/
https://redmine.openinfosecfoundation.org/issues/8863

### CODE REVIEW

File: tests/firewall/ruletype-firewall-144-bypass-keyword-drop/suricata.yaml [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    SSH_PORTS: 22
+    DNP3_PORTS: 20000
+    MODBUS_PORTS: 502
+    FILE_DATA_PORTS: "[$HTTP_PORTS,110,143]"
+    FTP_PORTS: 21
+    GENEVE_PORTS: 6081
+    VXLAN_PORTS: 4789
+    TEREDO_PORTS: 3544
+    SIP_PORTS: "[5060, 5061]"
+
```
Comment (jufajardini) [inline tests/firewall/ruletype-firewall-144-bypass-keyword-drop/suricata.yaml:43]: Can drop this part for probably all the config files in these tests.

**Comment by jufajardini** [comment]
Replaced by https://github.com/OISF/suricata-verify/pull/3271
